File: DataStructures/Set.py

class Set1(object):

    def __init__(self):

        self.items = []

    def add(self, item):

        if not item in self.items:

            self.items.append(item)

    def remove(self, item):

        self.items.remove(item)

    def __contains__(self, item):

        return item in self.items

class Set2(object):

    def __init__(self):

        self.items = set()

    def add(self, item):

        self.items.add(item)

    def remove(self, item):

        self.items.discard(item)

    def __contains__(self, item):

        return item in self.items

def main():

    for set_class in [Set1, Set2]:

        set_ = set_class()

        set_.add(1)
        set_.add(2)
        set_.add(1)

        print(1 in set_)

        set_.remove(1)

        print(1 in set_)

        print(2 in set_)

        print("------")

if __name__ == '__main__':
    main()

File: DataStructures/Set.java

import java.util.ArrayList;

public class Set {

    private interface SetDataStructure {

        public void add(int item);

        public boolean contains(int item);

        public void remove(int item);

    }

    private static class Set1 implements SetDataStructure {

        ArrayList<Integer> items;

        public Set1() {

            items = new ArrayList<>();

        }

        public void add(int item) {

            if (!items.contains(item)) {

                items.add(item);

            }

        }

        public boolean contains(int item) {

            return items.contains(item);

        }

        public void remove(int item) {

            items.remove(new Integer(item)); // Force object method

        }

    }

    private static class Set2 implements SetDataStructure {

        java.util.Set<Integer> items;

        public Set2() {

            items = new java.util.HashSet<>();

        }

        public void add(int item) {

            items.add(item);

        }

        public boolean contains(int item) {

            return items.contains(item);

        }

        public void remove(int item) {

            items.remove(item);

        }

    }

    public static void main(String[] args) {

        SetDataStructure[] setClasses = new SetDataStructure[]{new Set1(), new Set2()};

        for (SetDataStructure set : setClasses) {

            set.add(1);
            set.add(2);
            set.add(1);

            System.out.println(set.contains(1));

            set.remove(1);

            System.out.println(set.contains(1));

            System.out.println(set.contains(2));

            System.out.println("------");

        }

    }

}
